io.openems.edge.core/src/io/openems/edge/core/sum/SumImpl.java

```diff
@@ -339,7 +339,7 @@ private void calculateChannelValues() {
 				essMaxApparentPowerSum, gridMaxActivePowerSum, productionMaxAcActivePowerSum));
 
 		Long enterTheSystem = TypeUtils.sum(essActiveDischargeEnergySum, gridBuyActiveEnergySum,
-				productionActiveEnergySum);
+				productionAcActiveEnergySum);
```

Comment:
Can you explain why you changed this?
Because now, with my setup (gridmeter, smmetricEss, EssDCCharger) I got no ConsumptionEnergy in the history anymore.
Do I have to change something in my setup to get this working again?

Reply:
I guess the problem is, that the blueplanet Hybrid 10 doesn't deliver Charge/Discharge Energy values...

Reply:
The logic for Energy needs to be the same as with Power. Only AC-side is interesting there. See the line above for calculating the Consumption power. https://github.com/OpenEMS/openems/blob/3647cbd7d8f56d617b6c7c26aa026cad50830539/io.openems.edge.core/src/io/openems/edge/core/sum/SumImpl.java#L331

Reply:
Here is what I do in BpEss now:

```
	private final CalculateEnergyFromPower calculateAcChargeEnergy = new CalculateEnergyFromPower(this,
			SymmetricEss.ChannelId.ACTIVE_CHARGE_ENERGY);
	private final CalculateEnergyFromPower calculateAcDischargeEnergy = new CalculateEnergyFromPower(this,
			SymmetricEss.ChannelId.ACTIVE_DISCHARGE_ENERGY);
	private final CalculateEnergyFromPower calculateDcChargeEnergy = new CalculateEnergyFromPower(this,
			HybridEss.ChannelId.DC_CHARGE_ENERGY);
	private final CalculateEnergyFromPower calculateDcDischargeEnergy = new CalculateEnergyFromPower(this,
			HybridEss.ChannelId.DC_DISCHARGE_ENERGY);
```

and later in `updateChannels()`

```
				activePower = GlobalUtils.roundToPowerPrecision(batteryPower - inverter.getPvPower()) * -1; // invert
				dcDischargePower = GlobalUtils.roundToPowerPrecision(batteryPower) * -1; // invert
```
...
```
		// Calculate AC Energy
		if (activePower == null) {
			// Not available
			this.calculateAcChargeEnergy.update(null);
			this.calculateAcDischargeEnergy.update(null);
		} else if (activePower > 0) {
			// Discharge
			this.calculateAcChargeEnergy.update(0);
			this.calculateAcDischargeEnergy.update(activePower);
		} else {
			// Charge
			this.calculateAcChargeEnergy.update(activePower * -1);
			this.calculateAcDischargeEnergy.update(0);
		}

		// Calculate DC Energy
		if (dcDischargePower == null) {
			// Not available
			this.calculateDcChargeEnergy.update(null);
			this.calculateDcDischargeEnergy.update(null);
		} else if (dcDischargePower > 0) {
			// Discharge
			this.calculateDcChargeEnergy.update(0);
			this.calculateDcDischargeEnergy.update(dcDischargePower);
		} else {
			// Charge
			this.calculateDcChargeEnergy.update(dcDischargePower * -1);
			this.calculateDcDischargeEnergy.update(0);
		}
```